### DISCUSSION TITLE
Is the receipts download broken?

### DISCUSSION THREAD

**Discussion by Tilted-Software** [etsy/open-api#1199] [Q&A]
We're receiving a new "Object" object in the json downloaded from the receipts endpoint:

{
	"Object": {
		"count": 322,
		"results": [
			{
				"receipt_id": 3187890005,
				"receipt_type": 0,
				"seller_user_id": 445662765,
				"seller_email": "marketplace@

**Comment by Gareth064** [comment]
Which endpoint are you calling exactly?

**Reply by Gareth064** [reply]
If your talking about the endpoint to get all receipts for a shop
https://openapi.etsy.com/v3/application/shops/{shop_id}/receipts

Then this is the raw result I get after calling it

![image](https://github.com/etsy/open-api/assets/22307223/7e740ce3-b96e-4cfe-9b7f-ae3d529b9d3d)

![image](https://github.com/etsy/open-api/assets/22307223/63e8825e-7049-43e4-9558-27dde4850d0c)

No extra Object node.

Is your example the direct raw JSON string before any parsing in your app?

**Reply by Tilted-Software** [reply]
Please see below, it's the receipts end point 

![image](https://github.com/etsy/open-api/assets/42214446/37a2eb29-4eb1-4344-a6d0-8c2e6ff2f886)

**Reply by Gareth064** [reply]
What's happening inside `ReadObcjecResoponseAsync()`?
Thats where you appear to be passing the raw response to Deserialize it into an `object` to then Serialize into a string again using Netwonsoft.

**Reply by Tilted-Software** [reply]
I think the "Object" is a red herring now - apologies, I don't usually use auto-generated code for these things and it looks like that's something it introduces, when I look at the raw response using Fiddler it looks good, however it still won't deserialize into my model - has anything changed?  I've got two sites unable to download orders due to this - they've been working fine for years.

**Reply by Gareth064** [reply]
Nothing has changed as far as we know.  Everything my end is works and deserializes fine. You'd hope that Etsy would tell us if they were planning to do :P

When you say its been working for "years"... we are definitely still talking about the new v3 API here and you not somehow using the old V2 API? :D

**Reply by Tilted-Software** [reply]
If I strip out everything but receipt_id it deserializes fine which I guess means one of my types is wrong in the model, time for a bit "hunt for the incorrect type" by un-commenting out properties.

**Reply by Gareth064** [reply]
Are you using a code generator to wrap the Etsy API and build all your models etc? If so, there are always issues with types here and there. I remember seeing an email from someone in Etsy, recently, saying that they were making some fields nullable instead of returning an empty something like am empty string etc. But my god I can't find that email anymore. That was quite recent.

I avoided Code generators and just instead I have built my own C# Wrapper library around the Etsy V3 API. Its still in a private repo for now, but intend to make it public and publish on NuGet in the future for others to use.

**Comment by Tilted-Software** [comment]
Perhaps I should've done the same with the wrapper, I've got a load of comments at the top of the auto-generated code explaining how I changed it to get the thing working,

Found the issue anyway, looks like they may have brought that 5th Feb breaking change forward a week or so - I have the payment_email property set to not allows nulls - turns out it can now be null, I don't use it anyway so not a big deal but would've been nice to have got a heads-up on that.
